index.js

```diff
@@ -91,24 +91,31 @@ const getChunkType = chunk => {
 
 const {toString: objectToString} = Object.prototype;
 
-const getBufferedData = (chunks, getContents, textDecoder, length) => {
+const getBufferedData = ({chunks, getContents, getSize, textDecoder, length}) => {
 	try {
 		return getContents(chunks, textDecoder, length);
 	} catch {
-		return truncateBufferedValue(chunks, getContents, textDecoder);
+		return truncateBufferedValue({chunks, getContents, getSize, textDecoder});
 	}
 };
 
 // If the input is larger than the maximum length for a string or a buffer,
 // it will fail. We retry it with increasingly smaller inputs, so that
 // `error.bufferedData` is still set, albeit with a truncated value, since that
 // is still useful for debugging.
-const truncateBufferedValue = (chunks, getContents, textDecoder) => {
+const truncateBufferedValue = ({chunks, getContents, getSize, textDecoder}) => {
 	let chunksCount = chunks.length;
 	do {
 		chunksCount = Math.floor(chunksCount / SPLIT_FACTOR);
+		const fewerChunks = chunks.slice(0, chunksCount);
+
+		let length = 0;
+		for (const chunk of fewerChunks) {
+			length += getSize(chunk);
+		}
```

Comment:
This is the actual fix: the `length` property is required by `getContentsAsArrayBuffer()`.